Source/SwiftLintBuiltInRules/Rules/Metrics/LineLengthRule.swift

```diff
@@ -24,149 +23,224 @@ struct LineLengthRule: Rule {
             }
 
-            if configuration.ignoresInterpolatedStrings &&
-                lineHasKinds(line: line,
-                             kinds: [.stringInterpolationAnchor],
-                             kindsByLine: syntaxKindsByLine.value) {
-                return nil
+            // Populate interpolatedStringLines if ignores_interpolated_strings is true
+            if configuration.ignoresInterpolatedStrings {
+                let interpVisitor = InterpolatedStringLineVisitor(locationConverter: locationConverter)
+                interpVisitor.walk(node)
+                interpolatedStringLines = interpVisitor.lines
             }
 
-            if configuration.ignoresMultilineStrings &&
-               lineIsMultilineString(line, file: file, syntaxKindsByLine: syntaxKindsByLine.value) {
-                return nil
+            // Populate commentOnlyLines if ignores_comments is true
+            if configuration.ignoresComments {
+                commentOnlyLines = findCommentOnlyLines(in: node, file: file, locationConverter: locationConverter)
             }
 
-            for pattern in configuration.excludedLinesPatterns where line.containsMatchingPattern(pattern) {
-                return nil
+            return .skipChildren // We'll do the main processing in visitPost
+        }
+
+        override func visitPost(_ node: SourceFileSyntax) {  // swiftlint:disable:this unused_parameter
+            let minLengthThreshold = configuration.params.map(\.value).min() ?? .max
+
+            for line in file.lines {
+                // Quick check to skip very short lines before expensive stripping
+                // `line.content.count` <= `line.range.length` is true.
+                // So, `check line.range.length` is larger than minimum parameter value.
+                // for avoiding using heavy `line.content.count`.
+                if line.range.length < minLengthThreshold {
+                    continue
+                }
+
+                // Apply ignore configurations
+                if configuration.ignoresFunctionDeclarations && functionDeclarationLines.contains(line.index) {
+                    continue
+                }
+                if configuration.ignoresComments && commentOnlyLines.contains(line.index) {
+                    continue
+                }
+                if configuration.ignoresInterpolatedStrings && interpolatedStringLines.contains(line.index) {
+                    continue
+                }
+                if configuration.ignoresMultilineStrings && multilineStringLines.contains(line.index) {
+                    continue
+                }
+                if configuration.excludedLinesPatterns.contains(where: {
+                    regex($0).firstMatch(in: line.content, range: line.content.fullNSRange) != nil
+                }) {
+                    continue
+                }
+
+                // String stripping logic (remains largely the same as it's string-based)
```

Comment:
That's helpful for review, but in a few days we might wonder what this is supposed to mean. 😅